File: server/Audi/Entities/ProductCategory.cs
using System.Collections.Generic;

namespace Audi.Entities
{
    public class ProductCategory
    {
        public int Id { get; set; }
        public string Language { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int? ParentId { get; set; }
        public ProductCategory Parent { get; set; }
        public bool IsDeleted { get; set; } = false;
        public ICollection<ProductCategory> Children { get; set; }
        public ICollection<Product> Products { get; set; }
    }
}

File: client/projects/data/src/lib/models/product-category.ts
import { Product } from './product';

export interface ProductCategory {
  id: number;
  name: string;
  description: string;
  parentId?: number;
  children: ProductCategory[];
  products: Product[];
  isTopLevel: boolean;
}

export type ProductCategoryWithoutProducts = Omit<ProductCategory, 'products'>;
